src/components/Upload/UploadManager.tsx

```diff
@@ -0,0 +1,235 @@
+import {
+  Dialog,
+  DialogContent,
+  IconButton,
+  makeStyles,
+  DialogTitle as MuiDialogTitle,
+  Paper,
+  PaperProps,
+  Typography,
+} from '@material-ui/core';
+import {
+  Close as CloseIcon,
+  Maximize as MaximizeIcon,
+  Minimize as MinimizeIcon,
+} from '@material-ui/icons';
+import clsx from 'clsx';
+import React, {
+  FC,
+  memo,
+  useCallback,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+import Draggable, { DraggableData, DraggableEvent } from 'react-draggable';
+import { useWindowSize } from 'react-use';
+import { useSession } from '../Session';
+import { useUploadManager } from './UploadContext';
+
+const PAPER_WIDTH = 480;
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  root: {
+    pointerEvents: 'none',
+  },
+  paper: {
+    margin: 0,
+    pointerEvents: 'auto',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: `${PAPER_WIDTH}px`,
+  },
+  titleContainer: {
+    cursor: 'move',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: spacing(1),
+    '&.collapsed': {
+      cursor: 'default',
+    },
+  },
+  title: {
+    cursor: 'move',
+    paddingLeft: spacing(1),
+    '&.collapsed': {
+      cursor: 'default',
+    },
+  },
+  titleButtons: {
+    marginLeft: 'auto',
+  },
+  titleButton: {},
+  contentContainer: {
+    padding: spacing(1),
+    '&.collapsed': {
+      display: 'none',
+    },
+  },
+  noUploadsText: {
+    color: palette.grey[400],
+  },
+}));
+
+const initialPosition = { x: null as null | number, y: null as null | number };
+
+const PaperComponent: FC<PaperProps & { isCollapsed: boolean }> = ({
+  ...props
+}) => {
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+  const [isDragging, setIsDragging] = useState(false);
+  const paperRef = useRef<HTMLDivElement>(null);
+  const positionRef = useRef(initialPosition);
+  const { isCollapsed, ...rest } = props;
+
+  const { width: windowWidth, height: windowHeight } = useWindowSize();
+  const managerSize = {
+    width: (paperRef.current?.offsetWidth ?? 0) + 24,
+    height: (paperRef.current?.offsetHeight ?? 0) + 24,
```

Comment:
What's the 24 here for? Margin to add spacing between the edge of the screen? Maybe that could just be applied to the paper outright? That would enforce the spacing when dragging which seems nice too.

Reply:
I'll have to look at this again. You're right that that's what it's for, and there was a reason I now can't remember for why I hard-coded this in here. But I'm not crazy about it

Reply:
Mmm offsetWidth/Height doesn't include margin. maybe we could wrap it again and use padding so we don't have to apply it ourselves?

Reply:
Okay, we still need these values in order to keep the spacing between the edge of the screen because `margin` isn't included in `offsetWidth` or `offsetHeight`, but now it's calculated dynamically instead of hard-coded